### DISCUSSION TITLE
finetuning glowtts with new dataset - interpreting tensorboard evolution

### DISCUSSION THREAD

**Discussion by jefke24** [coqui-ai/TTS#810] [General Q&A]
Hello,

I'm just wondering if the behaviour I'm seeing is normal in tensorboard.
I'm new at this and I'm finetuning the glowtts model based on a new dataset, following the instructions here https://tts.readthedocs.io/en/latest/finetuning.html (nice job by the way in adding these instructions recently, just what I was looking for).

At first, in the first few epochs, I'm seeing this in tensorboard regarding groundtruth and prediction:

![image](https://user-images.githubusercontent.com/90794131/133487422-2bf63868-47c8-456e-aee8-203b39b5ac5f.png)

So at least I'm kind of recognizing the groundtruth in the prediction graph.
However, after about 150 epochs with 11 steps each, I end up with this:

![image](https://user-images.githubusercontent.com/90794131/133487551-9d72d7a5-685c-4eaf-b6ba-9c67cc3255f5.png)

Obviously the audio is complete noise at this point.

As for evolution of loss etc.:

![image](https://user-images.githubusercontent.com/90794131/133487666-2ac7b70c-a8ab-4e34-a17a-896722f427ee.png)

So if I'm reading this correctly, loss is going down and converging (however keeps getting stuck around 1.0, not going down anymore).
Attention graph is going towards a diagonal, which is the desired outcome I believe?

But prediction and ground truth are completely deviating.
I'm wondering: is this expected behaviour when finetuning a pre-trained model?
With enough training/epochs, can I expect groundtruth and prediction to converge again?

Or as soon as you're seeing this kind of behaviour, it's basically done and you need to start over with other parameters and/or better data?

Thanks in advance for any feedback you can give me!

**Comment by erogol** [comment]
It looks like there is something wrong internally, looking to the model outputs. Let me try training a GlowTTS myself and get back to you.

**Reply by jefke24** [reply]
ok, thanks @erogol, much appreciated!

**Comment by erogol** [comment]
btw your dataset look quite noisy  :)

**Reply by jefke24** [reply]
Well, like I said, first time for me doing this, I have no idea if the behaviour I'm seeing is because of my dataset, my config, ... or just normal behaviour when finetuning a pretrained model :)

If it helps, I can share you my config too?
Have been playing a lot with the learning rate, the signal normalization, sound normalization, mel fmin/fmax, ... but every time I'm seeing this kind of behaviour.

**Comment by erogol** [comment]
you are using the latest version right? v0.3.0

**Reply by jefke24** [reply]
Ah no, I believe I'm still using the previous release.
Didn't realize there was a more recent one.

Any signifcant changes regarding glowtss training in the new release then?

**Reply by erogol** [reply]
I don't remember but better to try the latest version. Also your config will be nice as you said.

**Reply by jefke24** [reply]
As requested, just redacted some param values which contain my name ;-)

`{
        "model": "glow_tts",
        "run_name": "testrun",
        "run_description": "glow-tts model training with relation position transformer encoder.",

        // AUDIO PARAMETERS
        "audio":{
            "fft_size": 1024,         // number of stft frequency levels. Size of the linear spectogram frame.
            "win_length": 1024,      // stft window length in ms.
            "hop_length": 256,       // stft window hop-lengh in ms.
            "frame_length_ms": null, // stft window length in ms.If null, 'win_length' is used.
            "frame_shift_ms": null,  // stft window hop-lengh in ms. If null, 'hop_length' is used.

            // Audio processing parameters
            "sample_rate": 22050,   // DATASET-RELATED: wav sample-rate. If different than the original data, it is resampled.
            "preemphasis": 0.99,     // pre-emphasis to reduce spec noise and make it more structured. If 0.0, no -pre-emphasis.
            "ref_level_db": 0,     // reference level db, theoretically 20db is the sound of air.

            "do_sound_norm": true,


            // Griffin-Lim
            "power": 1.2,           // value to sharpen wav signals after GL algorithm.
            "griffin_lim_iters": 60, // #griffin-lim iterations. 30-60 is a good range. Larger the value, slower the generation.

            // Silence trimming
            "do_trim_silence": true, // enable trimming of slience of audio as you load it. LJspeech (false), TWEB (false), Nancy (true)
            "trim_db": 60,          // threshold for timming silence. Set this according to your dataset.

            // MelSpectrogram parameters
            "num_mels": 80,         // size of the mel spec frame.
            "mel_fmin": 50.0,        // minimum freq level for mel-spec. ~50 for male and ~95 for female voices. Tune for dataset!!
            "mel_fmax": 8400.0,     // maximum freq level for mel-spec. Tune for dataset!!
            "spec_gain": 20.0,         // scaler value appplied after log transform of spectrogram.00

            // Normalization parameters
            "signal_norm": true,   // normalize spec values. Mean-Var normalization if 'stats_path' is defined otherwise range normalization defined by the other params.
            "min_level_db": -100,   // lower bound for normalization
            "symmetric_norm": true, // move normalization to range [-1, 1]
            "max_norm": 10.0,        // scale normalization to range [-max_norm, max_norm] or [0, max_norm]
            "clip_norm": true,      // clip normalized values into the range.
            "stats_path": null    // DO NOT USE WITH MULTI_SPEAKER MODEL. scaler stats file computed by 'compute_statistics.py'. If it is defined, mean-std based notmalization is used and other normalization params are ignored
        },

        // VOCABULARY PARAMETERS
        // if custom character set is not defined,
        // default set in symbols.py is used
        "characters": {
            "pad": "_",
            "eos": "~",
            "bos": "^",
            "characters": "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz!'(),-.:;? ",
            "punctuations": "!'(),-.:;? ",
            "phonemes": "abcdefhijklmnopqrstuvwxyz\u00e6\u00e7\u00f0\u00f8\u0127\u014b\u0153\u01c0\u01c1\u01c2\u01c3\u0250\u0251\u0252\u0253\u0254\u0255\u0256\u0257\u0258\u0259\u025a\u025b\u025c\u025e\u025f\u0260\u0261\u0262\u0263\u0264\u0265\u0266\u0267\u0268\u026a\u026b\u026c\u026d\u026e\u026f\u0270\u0271\u0272\u0273\u0274\u0275\u0276\u0278\u0279\u027a\u027b\u027d\u027e\u0280\u0281\u0282\u0283\u0284\u0288\u0289\u028a\u028b\u028c\u028d\u028e\u028f\u0290\u0291\u0292\u0294\u0295\u0298\u0299\u029b\u029c\u029d\u029f\u02a1\u02a2\u02c8\u02cc\u02d0\u02d1\u02de\u03b2\u03b8\u03c7\u1d7b\u2c71"
        },

        "add_blank": false, // if true add a new token after each token of the sentence. This increases the size of the input sequence, but has considerably improved the prosody of the GlowTTS model.

        // DISTRIBUTED TRAINING
        "distributed":{
            "backend": "nccl",
            "url": "tcp:\/\/localhost:54321"
        },

        "reinit_layers": [],    // give a list of layer names to restore from the given checkpoint. If not defined, it reloads all heuristically matching layers.

        // MODEL PARAMETERS
        // "use_mas": false,       // use Monotonic Alignment Search if true. Otherwise use pre-computed attention alignments.
        "hidden_channels_encoder": 192,
        "hidden_channels_decoder": 192,
        "hidden_channels_duration_predictor": 256,
        "use_encoder_prenet": true,
        "encoder_type": "rel_pos_transformer",
        "encoder_params": {
            "kernel_size":3,
            "dropout_p": 0.1,
            "num_layers": 6,
            "num_heads": 2,
            "hidden_channels_ffn": 768,
            "input_length": null
        },

        // TRAINING
        "batch_size": 16,       // Batch size for training. Lower values than 32 might cause hard to learn attention. It is overwritten by 'gradual_training'.
        "eval_batch_size":10,
        "r": 1,                 // Number of decoder frames to predict per iteration. Set the initial values if gradual training is enabled.
        "loss_masking": true,   // enable / disable loss masking against the sequence padding.
        "mixed_precision": true,
        "data_dep_init_iter": 10,

        // VALIDATION
        "run_eval": true,
        "test_delay_epochs": 0,       //Until attention is aligned, testing only wastes computation time.
        "test_sentences_file": null,  // set a file to load sentences to be used for testing. If it is null then we use default english sentences.

        // OPTIMIZER
        "noam_schedule": true,         // use noam warmup and lr schedule.
        "grad_clip": 5.0,              // upper limit for gradients for clipping.
        "epochs": 150,               // total number of epochs to train.
        "lr": 0.00001,                    // Initial learning rate. If Noam decay is active, maximum learning rate.
        "wd": 0.000001,                // Weight decay weight.
        "warmup_steps": 4000,          // Noam decay steps to increase the learning rate from 0 to "lr"
        "seq_len_norm": false,         // Normalize eash sample loss with its length to alleviate imbalanced datasets. Use it if your dataset is small or has skewed distribution of sequence lengths.

        // TENSORBOARD and LOGGING
        "print_step": 50,       // Number of steps to log training on console.
        "tb_plot_step": 100,    // Number of steps to plot TB training figures.
        "print_eval": false,     // If True, it prints intermediate loss values in evalulation.
        "save_step": 50,      // Number of training steps expected to save traninpg stats and checkpoints.
        "checkpoint": true,     // If true, it saves checkpoints per "save_step"
        "tb_model_param_stats": false,     // true, plots param stats per layer on tensorboard. Might be memory consuming, but good for debugging.

        // DATA LOADING
        "text_cleaner": "phoneme_cleaners",
        "enable_eos_bos_chars": false, // enable/disable beginning of sentence and end of sentence chars.
        "num_loader_workers": 4,        // number of training data loader processes. Don't set it too big. 4-8 are good values.
        "num_val_loader_workers": 4,    // number of evaluation data loader processes.
        "batch_group_size": 0,  //Number of batches to shuffle after bucketing.
        "min_seq_len": 3,       // DATASET-RELATED: minimum text length to use in training
        "max_seq_len": 500,     // DATASET-RELATED: maximum text length
        "compute_f0": false,     // compute f0 values in data-loader
        "use_noise_augment": true,  //add a random noise to audio signal for augmentation at training .
        "compute_input_seq_cache": true,

        // PATHS
        "output_path": "redacted",

        // PHONEMES
        "phoneme_cache_path": "redacted",  // phoneme computation is slow, therefore, it caches results in the given folder.
        "use_phonemes": true,           // use phonemes instead of raw characters. It is suggested for better pronounciation.
        "phoneme_language": "en-us",     // depending on your target language, pick one from  https://github.com/bootphon/phonemizer#languages

        // MULTI-SPEAKER and GST
        "use_speaker_embedding": false,     // use speaker embedding to enable multi-speaker learning.
        "use_external_speaker_embedding_file": false,
        "style_wav_for_test": null,          // path to style wav file to be used in TacotronGST inference.
        "use_gst": false,       // TACOTRON ONLY: use global style tokens

        // DATASETS
        "datasets":   // List of datasets. They all merged and they get different speaker_ids.
            [
                {
                    "name": "testrun",
                    "path": "redacted",
                    "meta_file_train": "metadata.csv",
                    "meta_file_val": null
                    // "path_for_attn": "/home/erogol/Data/LJSpeech-1.1/alignments/"
                }
            ]
    }

`

Am already running a new training from scratch with the latest release, just pulled from master.

Current first checkpoint status:

![image](https://user-images.githubusercontent.com/90794131/133586068-d3021ee9-f3ba-42f0-841f-1bf6182b5cfb.png)

By the way, to confirm: my audio samples are indeed at a sample rate of 22050, I made sure of that when I created them using audacity.

**Reply by jefke24** [reply]
Before I forget:
I'm using as pretrained model the model "tts_models/en/ljspeech/glow-tts".

I wrote my own formatter, but really simpel basically:
expects input to be filename.wav|transcription, grabs the transcription from column 1 after tokenizing on the "|" character, nothing more fancy than that.

**Reply by jefke24** [reply]
One more thing I can share: every time when a checkpoint is created, I also get this warning:
warning: audio amplitude out of range, auto clipped.

**Reply by erogol** [reply]
This is because the model output is out of range. that is also why the output spectrograms look like a smudge on a wall :)

**Reply by jefke24** [reply]
Huh, ok :)
So is this under my control, meaning I can get around this with config?

**Reply by erogol** [reply]
Can you try using the LJspeech glow-tts recipe for training your model? 

The config above is also very old. Or you can use the version that trained the released glow-tts model. Apparently, it was trained with v0.0.9

**Reply by jefke24** [reply]
But I guess, looking at the basic config there, that this is a recipe for a "train from scratch", not a finetuning?
So I'll need to add the restore path via the TrainingsArgs object, finetune the lr in the glowttsconfig object etc.

So, after having done that => won't I end up with basically the exact same config as now?
I see e.g. that use_phonemes is false in the recipe, but it's true in the config that came with the pretrained model.
So what do I use then?

Can I use the recipe with the pretrained released model, or do I need to switch to the v0.0.9 anyways if I want to use the pretrained model?
I don't have hours and hours of data, so I'm looking specifically for a case where I can finetune a pretrained model, not train one from scratch.

Edit: or are there any more recent pretrained models available?
I see this as a definite drawback => needing to switch to sometimes very old releases (I just noticed that 0.0.9 is even the first official release of Coqui) or needing to look for other models?
Surely there must be a way around this which doesn't require 1) users constantly switching to any old release, missing on all the bugs that have been fixed since then and features which have been added since then or 2) you guys having to retrain all released models for every new release of Coqui or 3) requiring the user to retrain from scratch every time there is a new release, thereby liquidating finetuning and forcing the user to have sufficient data for a train from scratch?

**Reply by jefke24** [reply]
Current recipe in use shared below.
1) does this seem correct to you?
2) from the first initial results of training, I have the impression it's still going in the same direction.


```
import os


from TTS.tts.configs import GlowTTSConfig


from TTS.tts.configs import BaseDatasetConfig
from TTS.config import BaseAudioConfig

from TTS.trainer import init_training, Trainer, TrainingArgs

output_path = "redacted"

dataset_config = BaseDatasetConfig(name="testrun", meta_file_train="metadata.csv", path=os.path.join(output_path, "redacted"))

audio_config = BaseAudioConfig(
    ref_level_db=0,
    preemphasis=0.99,
    power=1.2,
    signal_norm=True,
    do_sound_norm=True,
    mel_fmin=50,
    mel_fmax=8400,
    spec_gain=20.0, 
    trim_db=60,
    do_trim_silence=True,
    max_norm=10.0,
)

config = GlowTTSConfig(
    audio=audio_config,
    batch_size=16,
    eval_batch_size=10,
    num_loader_workers=4,
    num_eval_loader_workers=4,
    run_eval=True,
    test_delay_epochs=-1,
    epochs=150,
    text_cleaner="english_cleaners",
    use_phonemes=False,
    phoneme_language="en-us",
    phoneme_cache_path=os.path.join(output_path, "phoneme_cache"),
    print_step=5,
    print_eval=True,
    mixed_precision=False,
    output_path=output_path,
    datasets=[dataset_config],
    save_step=50,
    keep_after=50,
    use_noise_augment=True,
    lr=0.00001,
)

restore_path="/redacted/git/TTS/pretrained_glowtts_2/model_file.pth.tar"
continue_path="" 

training_args = TrainingArgs(restore_path=restore_path, continue_path=continue_path)


args, config, output_path, _, _, _= init_training(training_args, config)

trainer = Trainer(args, config, output_path)


trainer.fit()
```

**Reply by jefke24** [reply]
resulting config:

> {
>     "model": "glow_tts",
>     "run_name": "coqui_tts",
>     "run_description": "",
>     "epochs": 150,
>     "batch_size": 16,
>     "eval_batch_size": 10,
>     "mixed_precision": false,
>     "scheduler_after_epoch": false,
>     "run_eval": true,
>     "test_delay_epochs": -1,
>     "print_eval": true,
>     "dashboard_logger": "tensorboard",
>     "print_step": 5,
>     "plot_step": 100,
>     "model_param_stats": false,
>     "project_name": null,
>     "log_model_step": null,
>     "wandb_entity": null,
>     "save_step": 50,
>     "checkpoint": true,
>     "keep_all_best": false,
>     "keep_after": 50,
>     "num_loader_workers": 4,
>     "num_eval_loader_workers": 4,
>     "use_noise_augment": true,
>     "output_path": "redacted",
>     "distributed_backend": "nccl",
>     "distributed_url": "tcp://localhost:54321",
>     "audio": {
>         "fft_size": 1024,
>         "win_length": 1024,
>         "hop_length": 256,
>         "frame_shift_ms": null,
>         "frame_length_ms": null,
>         "stft_pad_mode": "reflect",
>         "sample_rate": 22050,
>         "resample": false,
>         "preemphasis": 0.99,
>         "ref_level_db": 0,
>         "do_sound_norm": true,
>         "log_func": "np.log10",
>         "do_trim_silence": true,
>         "trim_db": 60,
>         "power": 1.2,
>         "griffin_lim_iters": 60,
>         "num_mels": 80,
>         "mel_fmin": 50,
>         "mel_fmax": 8400,
>         "spec_gain": 20.0,
>         "do_amp_to_db_linear": true,
>         "do_amp_to_db_mel": true,
>         "signal_norm": true,
>         "min_level_db": -100,
>         "symmetric_norm": true,
>         "max_norm": 10.0,
>         "clip_norm": true,
>         "stats_path": null
>     },
>     "use_phonemes": false,
>     "use_espeak_phonemes": true,
>     "phoneme_language": "en-us",
>     "compute_input_seq_cache": false,
>     "text_cleaner": "english_cleaners",
>     "enable_eos_bos_chars": false,
>     "test_sentences_file": "",
>     "phoneme_cache_path": "redacted",
>     "characters": {
>         "pad": "_",
>         "eos": "~",
>         "bos": "^",
>         "characters": "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz!'(),-.:;? ",
>         "punctuations": "!'(),-.:;? ",
>         "phonemes": "iy\u0268\u0289\u026fu\u026a\u028f\u028ae\u00f8\u0258\u0259\u0275\u0264o\u025b\u0153\u025c\u025e\u028c\u0254\u00e6\u0250a\u0276\u0251\u0252\u1d7b\u0298\u0253\u01c0\u0257\u01c3\u0284\u01c2\u0260\u01c1\u029bpbtd\u0288\u0256c\u025fk\u0261q\u0262\u0294\u0274\u014b\u0272\u0273n\u0271m\u0299r\u0280\u2c71\u027e\u027d\u0278\u03b2fv\u03b8\u00f0sz\u0283\u0292\u0282\u0290\u00e7\u029dx\u0263\u03c7\u0281\u0127\u0295h\u0266\u026c\u026e\u028b\u0279\u027bj\u0270l\u026d\u028e\u029f\u02c8\u02cc\u02d0\u02d1\u028dw\u0265\u029c\u02a2\u02a1\u0255\u0291\u027a\u0267\u02b2\u025a\u02de\u026b"
>     },
>     "batch_group_size": 0,
>     "loss_masking": null,
>     "sort_by_audio_len": false,
>     "min_seq_len": 3,
>     "max_seq_len": 500,
>     "compute_f0": false,
>     "compute_linear_spec": false,
>     "add_blank": false,
>     "datasets": [
>         {
>             "name": "testrun",
>             "path": "redacted",
>             "meta_file_train": "metadata.csv",
>             "ununsed_speakers": null,
>             "meta_file_val": "",
>             "meta_file_attn_mask": ""
>         }
>     ],
>     "optimizer": "RAdam",
>     "optimizer_params": {
>         "betas": [
>             0.9,
>             0.998
>         ],
>         "weight_decay": 1e-06
>     },
>     "lr_scheduler": "NoamLR",
>     "lr_scheduler_params": {
>         "warmup_steps": 4000
>     },
>     "test_sentences": [],
>     "use_speaker_embedding": false,
>     "use_d_vector_file": false,
>     "d_vector_dim": 0,
>     "num_chars": null,
>     "encoder_type": "rel_pos_transformer",
>     "encoder_params": {
>         "kernel_size": 3,
>         "dropout_p": 0.1,
>         "num_layers": 6,
>         "num_heads": 2,
>         "hidden_channels_ffn": 768,
>         "input_length": null
>     },
>     "use_encoder_prenet": true,
>     "hidden_channels_enc": 192,
>     "hidden_channels_dec": 192,
>     "hidden_channels_dp": 256,
>     "dropout_p_dp": 0.1,
>     "dropout_p_dec": 0.05,
>     "mean_only": true,
>     "out_channels": 80,
>     "num_flow_blocks_dec": 12,
>     "inference_noise_scale": 0.0,
>     "kernel_size_dec": 5,
>     "dilation_rate": 1,
>     "num_block_layers": 4,
>     "num_speakers": 0,
>     "c_in_channels": 0,
>     "num_splits": 4,
>     "num_squeeze": 2,
>     "sigmoid_scale": false,
>     "data_dep_init_steps": 10,
>     "style_wav_for_test": null,
>     "length_scale": 1.0,
>     "d_vector_file": false,
>     "grad_clip": 5.0,
>     "lr": 1e-05,
>     "r": 1
> }

**Reply by erogol** [reply]
OK I figured the problem. It was internal. But it did not affect the model training, just the logging on the tensorboard. So your model should be okay. You can keep training until I release the fix. 

I also keep training my model with the fix and see if it works alright

**Reply by jefke24** [reply]
@erogol so basically it doesn't matter, using the recipe or using the "old config" that comes with the pretrained model, both will work taking into account your fix, using the latest version?

IF it works of course :)

**Reply by erogol** [reply]
Things fixed here https://github.com/coqui-ai/TTS/tree/fix_glow_tts. I can verify that it works.

**Reply by jefke24** [reply]
@erogol what do you recommend now?
1) As I was doing initially, using the old config of the pretrained model with the necessary adjustments and then training with release 0.3.0?
2) Training using the recipe, using release 0.3.0?
3) using the old config of the pretrained model with the necessary adjustments but using release 0.0.9?

**Reply by jefke24** [reply]
There is also something else I was wondering: if you look at my config, I configured the learning rate as 0.00001.
However, when I look at the output of my training steps, I see this:

![image](https://user-images.githubusercontent.com/90794131/133736275-c60a3580-93b1-436f-93ed-db957e31a383.png)

As far as I am aware, 0.00001 != 0.00011 :)

Is this also normal behaviour?

**Reply by erogol** [reply]
It is using Noam scheduling so it attenuates

**Reply by erogol** [reply]
> @erogol what do you recommend now?
> 
> 1. As I was doing initially, using the old config of the pretrained model with the necessary adjustments and then training with release 0.3.0?
> 2. Training using the recipe, using release 0.3.0?
> 3. using the old config of the pretrained model with the necessary adjustments but using release 0.0.9?

Just use the LJSpeech recipe and edit the config for your dataset

**Reply by jefke24** [reply]
So just for my correct understanding, when you say use the LJSpeech recipe and edit the config for your dataset, you ARE saying this is also the valid approach for finetuning with a more limited dataset using the pretrained glowtts model, which was trained with v0.0.9 right? :)

Just wanting to make sure we're talking about the same here, finetuning, not learning a new model from scratch :)

**Reply by erogol** [reply]
Yes you can still use the recipe and apply this https://tts.readthedocs.io/en/latest/finetuning.html

**Reply by jefke24** [reply]
Alright, thanks for the effort and for fixing this!

**Comment by jefke24** [comment]
@erogol, I pulled in your commit and continued training.
I'm definitely seeing new graphs. 

I see you also added some pre-defined test-sentences, would explain the additional graphs I'm seeing now.

However, regarding the training graphs, is this normal?

![image](https://user-images.githubusercontent.com/90794131/133891655-22d0cb45-a1be-4a23-9a7d-9ddd78ab9ce6.png)

The fact that the ground truth has this "bar" on the right, but the prediction doesn't have this?
Seems almost as if the ground truth and prediction are for different samples?

Edit:
the same I think for the eval figures.
![image](https://user-images.githubusercontent.com/90794131/133891936-c734723e-9b6c-4504-a030-8cb46249c3a2.png)

Could be my misinterpretation, but I have the impression these 2 don't match either?
There's no bar here, but the darker and lighter areas don't seem to match I think?

**Reply by erogol** [reply]
I suspect it is about your dataset. Either the transcripts and the audios do not match or the audio quality is too bad for the model to learn.

**Reply by jefke24** [reply]
Strange.
I'm pretty sure the transcripts are good.

Before you made the change, I did have a 100% match on every graph: if the ground_truth had a bar on the right, the prediction also had one and always one of the same dimensions, was an exact match every time.

For the fun of it, I also started training a VITS model this weekend with the exact same dataset and I'm not seeing any inconsistencies there at all in any graph.

**Reply by erogol** [reply]
After the change, I was able to train a GlowTTS model using the LJSpeech recipe so I am sure it works for the LJSpeech dataset. So your dataset format is the only variable here that I am not sure.  How did you format your dataset and the metafile ?
